### PR TITLE
Windows: Fix `sepfir2d` tests

### PR DESCRIPTION
**Pull request by leofang**
```
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_0_{hcol=1, hrow=1, input=(256, 256)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_1_{hcol=1, hrow=1, input=(4, 512)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_2_{hcol=1, hrow=1, input=(512, 3)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_3_{hcol=1, hrow=3, input=(256, 256)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_4_{hcol=1, hrow=3, input=(4, 512)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_5_{hcol=1, hrow=3, input=(512, 3)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_6_{hcol=3, hrow=1, input=(256, 256)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_7_{hcol=3, hrow=1, input=(4, 512)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_8_{hcol=3, hrow=1, input=(512, 3)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_9_{hcol=3, hrow=3, input=(256, 256)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_10_{hcol=3, hrow=3, input=(4, 512)}::test_sepfir2d
FAILED tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py::TestSepFIR2d_param_11_{hcol=3, hrow=3, input=(512, 3)}::test_sepfir2d
```

### CODE REVIEW

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
Wait, the Windows CI is already on SciPy 1.6.1?!

**Comment by leofang** [comment]
```
____ TestSepFIR2d_param_0_{hcol=1, hrow=1, input=(256, 256)}.test_sepfir2d ____

self = <<test_bsplines.TestSepFIR2d_param_0_{hcol=1, hrow=1, input=(256, 256)} testMethod=test_sepfir2d>  parameter: {'hcol': 1, 'hrow': 1, 'input': (256, 256)}>
xp = <module 'numpy' from 'C:\\Development\\Python\\Python37\\lib\\site-packages\\numpy\\__init__.py'>
scp = <module 'scipy' from 'C:\\Development\\Python\\Python37\\lib\\site-packages\\scipy\\__init__.py'>
dtype = <class 'numpy.complex64'>

    @testing.for_all_dtypes()
    @testing.numpy_cupy_allclose(atol=1e-5, rtol=1e-5, scipy_name='scp')
    def test_sepfir2d(self, xp, scp, dtype):
        input = testing.shaped_random(self.input, xp, dtype)
        hrow = testing.shaped_random((self.hrow,), xp, dtype)
        hcol = testing.shaped_random((self.hcol,), xp, dtype)
>       return scp.signal.sepfir2d(input, hrow, hcol)
E       AssertionError: Only numpy raises error
E       
E       Traceback (most recent call last):
E         File "C:\Windows\TEMP\flexci\run-00142981\work\src\cupy\testing\helper.py", line 47, in _call_func
E           result = impl(self, *args, **kw)
E         File "C:\Windows\TEMP\flexci\run-00142981\work\src\tests\cupyx_tests\scipy_tests\signal_tests\test_bsplines.py", line 27, in test_sepfir2d
E           return scp.signal.sepfir2d(input, hrow, hcol)
E       ValueError: Incorrect type.

tests\cupyx_tests\scipy_tests\signal_tests\test_bsplines.py:27: AssertionError
---------------------------- Captured stdout call -----------------------------
dtype is <class 'numpy.complex64'>
```

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
No luck...I don't have resource to look into this today. Maybe tomorrow...

File: tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py [hunk 1]
```diff
@@ -18,9 +19,13 @@
 @testing.gpu
 @testing.with_requires('scipy')
 class TestSepFIR2d(unittest.TestCase):
+
     @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(atol=1e-5, rtol=1e-5, scipy_name='scp')
     def test_sepfir2d(self, xp, scp, dtype):
+        if sys.platform.startswith('win32') and xp.dtype(dtype).kind in 'iu':
```
Comment (emcastillo) [inline tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py:26]:
```suggestion
        if sys.platform.startswith('win32') and xp.dtype(dtype).kind in 'c':
```

**Comment by leofang** [comment]
Jenkins, test this please

**Review by emcastillo — APPROVED** [review]

**Comment by leofang** [comment]
Test failures are resolved:
https://storage.googleapis.com/chainer-artifacts-pfn-public-ci/cupy-ci/69462/cupy_test_log.txt

**Comment by leofang** [comment]
Reported the SciPy error to upstream (https://github.com/scipy/scipy/issues/13643).

**Comment by emcastillo** [comment]
v8 does not need this as it doesn't use complex
https://github.com/cupy/cupy/blob/v8/tests/cupyx_tests/scipy_tests/signal_tests/test_bsplines.py#L21

**Comment by leofang** [comment]
> v8 does not need this as it doesn't use complex

Ah great! Thanks, @emcastillo, I forgot about it.

**Pull request merged**